Refactor board region container

eferos93 opened the pull request:
While working on UI, I tried to refactor Board and RegionContainer, by moving some responsibilities from the second to the first. They look more balanced to me now and board now is not a bloater anymore. 
Let me know what do you think.

Commit:
moved two methods from Board to RegionContainer

Commit:
remove not useful tests; rename method in Chain; removed method in Board

Commit:
moved methods to RegionContainer

Commit:
remove methods in Board

Commit:
rename few methods

Commit:
added commented code, I'll finish tomorrow

Commit:
removed code smell: extracted method

Commit:
added TODO

demirbilek95 reviewed (approved):
I think it seems better

Commit:
refactor BoardSideShould.checkAdjacentPositionToSides

Commit:
refactor test BoardShould.checkDiagonalAdjacent

Commit:
refactor BoardShould.checkOrthogonalAdjacent

Commit:
removed usage of \t in print board: might lead to bad interoperability between different systems

Commit:
rename class Chain to ChainContainer

Commit:
refactored ChainContainer: moved some responsabilities from Board to ChainContainer

Commit:
solved feature envy problem: moved field "sides" to ChainContainer as it is useful only in this class

Commit:
remove singleton pattern in RegionContainer, made constructor package private;
Remove Class RegionContainerShould, those test were accessing functions that are now private; Region container functionality is tested via method BoardShould.fillTerritoryWithEqualNumberOfStoneOfTheSameColor and BoardShould.fillTerritoryWithDifferentNumberOfStone

eferos93 commented:
Yo, take a final look so then we merge thiz

wilsonjefferson commented on src/test/java/quentin/core/ChainContainerShould.java:
```diff
@@ -2,13 +2,13 @@
+public class ChainContainerShould {
 
     @Test
     public void touchEdgesIfItIsACompleteChain() {
```
ChainContainerShould.touch`Edges`IfItIsACompleteChain should be touch`Sides`IfItIsACompleteChain.

wilsonjefferson commented on src/test/java/quentin/core/ChainContainerShould.java:
```diff
@@ -19,15 +19,9 @@ public void touchEdgesIfItIsACompleteChain() {
+        assertEquals(Stone.BLACK, chainContainer.getColorWithCompleteChain());
     }
 
     @Test
```
same here: form edges to sides.

Commit:
rename methods

wilsonjefferson commented:
For me it is okay.
